File: 10/clojure/src/Owl.clj
;;;; Owl.clj provides an Owl "class".
;;;; 
;;;; Begun by: Prof. Adams, for CS 214 at Calvin College.
;;;; Completed by:
;;;; Date:
;;;;
;;;; Bring in 'Bird' (to be used as superclass) 

(load "Bird")

;; define 'Owl' as 'subclass' of Bird


(defrecord Owl [^Bird name] )

;;; constructors
;;; - default: takes no arguments
;;;   Postcondition: name is initialized to a default value
;;; - explicit: takes one parameter
;;;   Receive: itsName, a String
;;;   Postcondition: name == itsName
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn make-Owl
  ([]                (->Owl "Ann Onymous"))
  ([^String itsName] (->Owl itsName))
  )

;;; method to retrieve a Owl's class
;;; Receive: this, a Owl object.
;;; Return: "Owl"
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defmethod getClass Owl [ _ ]
  "Owl"
  )

;;; method to retrieve a Owl's call
;;; Receive: this, a Owl object.
;;; Return: "WHOOOOO"
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defmethod getCall Owl [ _ ]
  "Whoooooo"
  )

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

File: 10/java/Owl.java
/* Owl.java provides an Owl subclass of Bird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by:
 * Date:
 ******************************************************/

public class Owl extends Bird {

	/* explicit constructor
	 * Receive: name, a String 
	 * PostCond: myName == name.
	 */

	public Owl()
	{
		super();
	}

	public Owl(String name)
	{
		super(name);
	}



	/* An Owl's Call
	 * Return: an owl-call ("Whoo-hoo!").
	 */

	public String call()
	{
		return "Whoo-hoo!";
	}

	public void print()
	{
		System.out.println( getName() + ' ' + getClass().getName() + " says " + call());
	}


} 

